### PR TITLE
Cyborgs can now choose what type of brain they spawn with.

### LINKED ISSUE #23582: IPC posi brains dont come out as a MMI for borgs.
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

515.1621

### Issue Description

If you're deconstructed as a borg and your character is a IPC, you pop out lacking a MMI and as a human brain, and it puts in big red bold text that somethings fucked up, please make report. 

### What did you expect to happen?

To come out as a posionic brain 

### What happened instead?

Flesh brain without a MMI

### Why is this bad/What are the consequences?

Not intended, big red bold text talking about bugs is not cool for ic.

### Steps to reproduce the issue.

have your character a ipc
become borg, probably only roundstart borg.
get deconstructed
be turned into a human brain

### When did the problem start happening?

Beats me.

### Extra information

_No response_

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by BiancaWilkson**
## What Does This PR Do

This PR adds a new section to character creation, allowing players to choose whether their borg will spawn in with an MMI, a robobrain, or with a positronic brain. It also adds a new column to the SQL database for storing this info.
Fixes #23582
## Why It's Good For The Game

It adds more options for character customization for a role that is already very difficult to customize a character for.
## Testing

Spawned in with all different types of borg, and gibbed them to make sure the brains were correct. I don't have SQL database set up myself, so I was unfortunately not able to test if the queries work properly. Advice is appreciated.
## Changelog
:cl:
add: Character creation now lets you choose what type of brain a cyborg will spawn with.
/:cl:

### CODE REVIEW

**Comment by 1080pCat** [comment]
this'd probably fix/close https://github.com/ParadiseSS13/Paradise/issues/23582

**Comment by Fordoxia** [comment]
Whilst you're rooting around in that dark corner of the code, might you also add an option to set cyborg flavour text in the creator (rather than having to manually set it with the verb every round)?

**Comment by DGamerL** [comment]
> Whilst you're rooting around in that dark corner of the code, might you also add an option to set cyborg flavour text in the creator (rather than having to manually set it with the verb every round)?

Very out of scope

**Review by DGamerL — CHANGES_REQUESTED** [review]
image

File: SQL/updates/53-54.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#Updating SQL from 53 to 54 -Wilk
+#Add a choice for what type of brain borgs want to have
+
+ALTER TABLE 'characters'
+	ADD COLUMN 'cyborg_brain_type' VARCHAR(11) NOT NULL DEFAULT 'MMI' AFTER 'height';
```
Comment (AffectedArc07) [inline SQL/updates/53-54.sql:5]:
```suggestion
ALTER TABLE `characters`
	ADD COLUMN `cyborg_brain_type` VARCHAR(11) NOT NULL DEFAULT 'MMI' AFTER `height`;
```

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]
Also you need to bump `#define SQL_VERSION` and the default version in the config.

**Review by lewcc — COMMENTED** [review]
Little things, mostly looks good!

File: SQL/paradise_schema.sql [hunk 2]
```diff
@@ -79,6 +79,7 @@ CREATE TABLE `characters` (
   `hair_gradient_colour` varchar(7) COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '#000000',
   `hair_gradient_alpha` tinyint(3) UNSIGNED NOT NULL DEFAULT '255',
   `custom_emotes` LONGTEXT COLLATE 'utf8mb4_unicode_ci' DEFAULT NULL,
+  `cyborg_brain_type` varchar(11) NOT NULL DEFAULT 'MMI',
```
Comment (AffectedArc07) [inline SQL/paradise_schema.sql:82]: Make this an enum in the DB for the 3 types, saves on disk space and parsing time.

File: code/_globalvars/lists/flavor_misc.dm [hunk 3]
```diff
@@ -61,3 +61,9 @@ GLOBAL_LIST_INIT(phonetic_alphabet, list("Alpha", "Bravo", "Charlie",
 #define DSATCHEL "Department Satchel"
 #define DDUFFLEBAG "Department Dufflebag"
 GLOBAL_LIST_INIT(backbaglist, list(DBACKPACK, DSATCHEL, DDUFFLEBAG, GBACKPACK, GSATCHEL, GDUFFLEBAG, LSATCHEL))
+
+//Chooseable brain types for borgs
+#define MMI_BORG "MMI"
+#define ROBOBRAIN_BORG "Robobrain"
+#define POSITRONIC_BORG "Positronic"
+GLOBAL_LIST_INIT(borg_brain_choices, list(MMI_BORG, ROBOBRAIN_BORG, POSITRONIC_BORG))
```
Comment (AffectedArc07) [inline code/_globalvars/lists/flavor_misc.dm:69]: Global protect this since it goes in the DB

File: code/modules/client/preference/character.dm [hunk 4]
```diff
@@ -551,7 +556,7 @@
 	loadout_gear = sanitize_json(loadout_gear)
 	custom_emotes_tmp = sanitize_json(custom_emotes_tmp)
 	custom_emotes = init_custom_emotes(custom_emotes_tmp)
-
+	cyborg_brain_type = sanitize_text(cyborg_brain_type, initial(cyborg_brain_type))
```
Comment (AffectedArc07) [inline code/modules/client/preference/character.dm:559]: This should be a sanitize inlist so that it fails if the type from the DB isn't valid.

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

**Review by AffectedArc07 — APPROVED** [review]

**Review by S34NW — APPROVED** [review]
Approved, must wait for AA to merge as SQL and config changes required

**Review by AffectedArc07 — APPROVED** [review]
# FULL SEND

File: SQL/updates/53-54.sql [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#Updating SQL from 53 to 54 -Wilk
+#Add a choice for what type of brain borgs want to have
+
+ALTER TABLE `characters`
+	ADD COLUMN `cyborg_brain_type` VARCHAR(11) NOT NULL DEFAULT 'MMI' AFTER `height`;
```
Comment (AffectedArc07) [inline SQL/updates/53-54.sql:5]: Oh fuck this should be the enum as well.

# FUCK IT FUTURE PR

**Comment by streather** [comment]
probably nothing, but if you open the dialog to select what type of brain and hit cancel, it'll set it to no brain.

would it default to a braintype if that box is blank?

**Comment by Fordoxia** [comment]
> probably nothing, but if you open the dialog to select what type of brain and hit cancel, it'll set it to no brain.
> 
> would it default to a braintype if that box is blank?

I'm pretty sure that even if no brain, there's a piece of code in robot_mob.dm that will give a brain to any borg that spawns without a brain because borgs without brains has happened before and is very bad.

**Pull request merged**